src/providers/gdal/qgsgdalprovider.h

```diff
@@ -158,6 +163,15 @@ class QgsGdalProvider : public QgsRasterDataProvider, QgsGdalProviderBase
     QString validatePyramidsConfigOptions( QgsRaster::RasterPyramidsFormat pyramidsFormat,
                                            const QStringList &configOptions, const QString &fileFormat ) override;
   private:
+
+    QgsGdalProvider( const QgsGdalProvider &other );
+
+    // reference counter to know how many main and cloned provider instances are linked
+    int *mpRefCounter = nullptr;
```

Comment:
Is a  QAtomicInt more appropriate here?

Reply:
should be a QAtomicInt* since we want the same counter to be shared by all clones. Anyway I already needed a mutex, so the use of the counter within the mutex is safe

Reply:
Ah, of course!

+1 to merge from me